asteroid/engine/schedulers.py

```diff
@@ -1,6 +1,9 @@
+from torch.optim.optimizer import Optimizer
 
-class _BaseScheduler(object):
+
+class _BaseScheduler(Optimizer):
     def __init__(self, optimizer):
+        super(_BaseScheduler, self).__init__(optimizer.__dict__["param_groups"], optimizer.__dict__["defaults"])
```

Comment:
this is hacky... because lightnining checks the type of optimizer, the scheduler should be an optimizer. This does it but it would be better if there is a way to directly wrap the optimizer by overriding its methods

Reply:
I think, this is possible, with [this](https://stackoverflow.com/a/46757134/7248396)

Reply:
I will go with this I think

```python
class _BaseScheduler(object):
    def __init__(self, optimizer):
        self.__class__ = type(optimizer.__class__.__name__,
                              (self.__class__, optimizer.__class__),
                              {})
        self.__dict__ = optimizer.__dict__
        self.optimizer = optimizer
        self.step_num = 0
```

Reply:
fails with distributed data parallel